### ISSUE TITLE
Where can i find the source code of SANet？

### ISSUE THREAD

**Issue by linqiaozhou** [gjy3035/Awesome-Crowd-Counting#27]
Does the SANet opened the source code ?

**Comment by gjy3035** [comment]
The authors of SANet do not release the code. I try to reproduce it  in C^3 Framework, which can achieve 12,1 MAE on Shanghai Tech B (far from 8.0 in the paper). Since C^3 Framework is still under development, I plan release it in March, 2019. Thanks for your attention.

**Comment by LaiPiXiong** [comment]
I only can reach 43.3575 on ShanghaiTechB (Pytorch, 500 epochs)

**Comment by gjy3035** [comment]
@LaiPiXiong According to my implemention, the SSIM loss may cause the network does not converge. Do you have this problem?

**Comment by Ethanwl** [comment]
@gjy3035 The SANet paper says that We use four convolutions to progressively refine the details of feature maps, with filter sizes from 9 × 9 to 3 × 3. However, there are five convolutions(9, 7, 5, 3, 5) on the picture but the 1x1 convolution is not on the picture. The author writes it wrong? Or is the picture wrong? How do you implement it? Thanks for your Awesome-Crowd-Counting.  I'm looking forward to the C^3 Framework.

**Comment by LaiPiXiong** [comment]
@gjy3035 We do not try training without SSIM, so we don't meet this problem. But their testing on Batches is really effective( MAE on ShanghaiB drops to 19.3).

**Comment by shriramhr** [comment]
> The authors of SANet do not release the code. I try to reproduce it in C^3 Framework, which can achieve 12,1 MAE on Shanghai Tech B (far from 8.0 in the paper). Since C^3 Framework is still under development, I plan release it in March, 2019. Thanks for your attention.

Hi, 
Any updates on this? I've been implementing jokermask's version of SANet and it works okay, not getting the best MAE but it's not bad. The SSIM loss has not been implemented. Looking forward to yours.

**Comment by gjy3035** [comment]
@shriramhr Due to many deadlines, som codes is still checked carefully, which will take 2 weeks.

**Comment by shriramhr** [comment]
> @shriramhr Due to many deadlines, som codes is still checked carefully, which will take 2 weeks.

Hi, 

I'm trying to implement it on my own on pytorch. Could you please tell me how you can implement SSIM index loss if you have already worked on this. I'm very confused on how to do it. Thank you!

**Comment by gjy3035** [comment]
The concrete SA code is available at https://github.com/gjy3035/C-3-Framework
